core/src/main/scala/org/apache/spark/storage/BlockId.scala

```diff
@@ -52,8 +52,9 @@ case class RDDBlockId(rddId: Int, splitIndex: Int) extends BlockId {
 // Format of the shuffle block ids (including data and index) should be kept in sync with
 // org.apache.spark.network.shuffle.ExternalShuffleBlockResolver#getBlockData().
 @DeveloperApi
-case class ShuffleBlockId(shuffleId: Int, mapId: Int, reduceId: Int) extends BlockId {
-  override def name: String = "shuffle_" + shuffleId + "_" + mapId + "_" + reduceId
+case class ShuffleBlockId(shuffleId: Int, mapId: Int, reduceId: Int, length: Int = 1)
+  extends BlockId {
+  override def name: String = "shuffle_" + shuffleId + "_" + mapId + "_" + reduceId + "_" + length
```

Comment:
nit: maybe `s"shuffle_$shuffleId_$mapId_$reduceId_$length"`?

Reply:
these are semi-public interfaces, can we create a new block id `ContinuousShuffleBlockIds`?

Reply:
`ContinuousShuffleBlockIds` looks like a good idea, let me try.